### PR TITLE
Add rent functionality

### PR DESCRIPTION
**Pull request by Cialkowski**
Description
added renting posibility with customer creation
Checklist for Developer:
 - [x] .cs issue
 - [ ] MySql issue
 - [ ] Document issue
 - [x] Add reviewers to pull-request
 - [x] Assign yourself to pull-request
 - [x] Add labels to pull-request
 - [ ] Link an issue to from repository
Please note that all items which are listed above are relevant to merge the pull-request!

If the checkboxes are empty pull-request will be closed!

### CODE REVIEW

File: library/library/Customer.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            Surrname = surrname;
+            PhoneNumber = phoneNumber;
+            Email = email;
+        }
+
+        public void AddCustomer()
+        {
+            using (var connection = new SqlConnection(DbCon.ConnectionString))
+            {
+                connection.Open();
```
Comment (Borzygniew) [inline library/library/Customer.cs:33]: where close

File: library/library/Rent.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                adapter.Update(table);
+            }
+            catch (Exception ex) { MessageBox.Show(ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error); }
+        }
+
+        public void UpdateCopyRent()
+        {
+            using (var connection = new SqlConnection(DbCon.ConnectionString))
+            {
+                connection.Open();
```
Comment (Borzygniew) [inline library/library/Rent.cs:55]: where close 2

File: library/library/RentForm.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        public int IdBook { get; }
+
+        public RentForm(int idBook)
+        {
+            IdBook = idBook;
+            InitializeComponent();
+        }
+
+        private void button1_Click(object sender, EventArgs e)
```
Comment (Borzygniew) [inline library/library/RentForm.cs]: please change default control name next time

**Review by Borzygniew — CHANGES_REQUESTED** [review]
need to check how it works and looks like
overall 🦐🦉

File: library/library/Customer.cs (hunk 1 above)
Comment (Cialkowski) [reply library/library/Customer.cs:33]: connection is used only in `using` code block. It closes connection itself

File: library/library/Rent.cs (hunk 2 above)
Comment (Cialkowski) [reply library/library/Rent.cs:55]: Same

File: library/library/RentForm.cs (hunk 3 above)
Comment (Cialkowski) [reply library/library/RentForm.cs]: Done

File: library/library/Customer.cs (hunk 1 above)
Comment (Borzygniew) [reply library/library/Customer.cs:33]: how?

Comment (Cialkowski) [reply library/library/Customer.cs:33]: closed connection

File: library/library/Rent.cs (hunk 2 above)
Comment (Cialkowski) [reply library/library/Rent.cs:55]: closed connection

File: library/library/Rent.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                        var command = new SqlCommand("INSERT INTO CopyRent (id_rent, id_copy) VALUES (@idRent, @idCopy)", connection, transaction);
+                        command.Parameters.AddWithValue("@idRent", GetLastRentId());
+                        command.Parameters.AddWithValue("@idCopy", IdCopy);
+                        command.ExecuteNonQuery();
+                        transaction.Commit();
+                    }
+                    catch (Exception ex)
+                    {
+                        transaction.Rollback();
+                        throw new Exception(ex.Message);
```
Comment (Borzygniew) [inline library/library/Rent.cs:69]: why you throw exception like that here?

File: library/library/Customer.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                        command.ExecuteNonQuery();
+                        transaction.Commit();
+                    }
+                    catch (Exception ex)
+                    {
+                        transaction.Rollback();
+                        throw new Exception("Failed to add customer: " + ex.Message);
+                    }
+                }
+                connection.Close();
```
Comment (Borzygniew) [inline library/library/Customer.cs:52]: one time closing looks like this:  `finally { connection.Close(); }`
another time is just empty `.Close()` after try catch

**Review by Borzygniew — CHANGES_REQUESTED** [review]
I still didn't test how it works anyway

**Review by Borzygniew — APPROVED** [review]
bydzie

**Pull request merged**